app/gui2/src/stores/project/index.ts

```diff
@@ -555,6 +554,22 @@ export const useProjectStore = defineStore('project', () => {
     )
   }
 
+  const dataflowErrors = new ReactiveMapping(computedValueRegistry.db, (id, info) =>
+    info.payload.type !== 'DataflowError'
+      ? undefined
+      : (useVisualizationData(
+          ref({
+            expressionId: id,
+            visualizationModule: 'Standard.Visualization.Preprocessor',
+            expression: {
+              module: 'Standard.Visualization.Preprocessor',
+              definedOnType: 'Standard.Visualization.Preprocessor',
+              name: 'error_preprocessor',
+            },
+          }),
+        ) as ShallowRef<{ kind: 'Dataflow'; message: string } | undefined>),
```

Comment:
That typecast is very sketchy. I'd prefer to validate the output shape with `zod`, so an unexpected API change/bug on the engine or libs side doesn't bite us here. Too many systems involved to just trust it blindly. I'd rather throw a big error in the console when that happens.